src/fenicsxconcrete/material.py

```diff
@@ -0,0 +1,19 @@
+
+from typing import Callable
+from pydantic.dataclasses import dataclass
+
```

Comment:
This is only a general definition of the material, right? At which point do we implement what happens if you choose e.g linear elasticity?

Reply:
At the level of your solver (FEMProblemDefinition -though the name is IMO too long). You would then built your weak form and have an additional configuration file (but that does not have to be general but is really depending on your problem) that includes the material parameters (if you always use linearElastic), or the constitutive law and the parameters (if you decide to be able to allow on the configuration file level different material laws).